### PR TITLE
IC-1334 add session progress section to the intervention progress page

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

Adds a 'session progress' section to the intervention progress page.

This PR also contains is package for parsing ISO8061 date time strings to human readable strings for presentation. 

## What is the intent behind these changes?

Allow SPs to view and update sessions for interventions.

Screenshot 2021-03-24 at 12 58 24

### CODE REVIEW

**Comment by lawrence-forooghian** [comment]
@tomsmyers I've converted this one to a draft until it's ready for review

File: server/utils/viewUtils.ts [hunk 1]
```diff
@@ -54,4 +63,8 @@ export default class ViewUtils {
       }),
     }
   }
+
+  static govukTag(text: string, colour?: GovukColour): string {
```
[the remark this answers is not in the document]
Comment (Gweaton) [reply server/utils/viewUtils.ts:67]: Or @lawrence-forooghian do you have any strong preferences for using this vs the other method?

**Review by Gweaton — COMMENTED** [review]
A couple of things, one of which we've just discussed over a call.

File: server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts [hunk 2]
```diff
@@ -96,7 +96,40 @@ export default class ServiceProviderReferralsController {
       serviceUserPromise,
     ])
 
-    const presenter = new InterventionProgressPresenter(sentReferral, serviceCategory, actionPlan, serviceUser)
+    let actionPlanAppointments: ActionPlanAppointment[] = []
+
+    // this code is horribly inefficient and results in a lot of unnecessary log messages from the client.
+    // we are going to change the backend so that all the appointments exist for a submitted action plan,
+    // which will allow us to use the `getActionPlanAppointments` method and clean up this code.
```
[the remark this answers is not in the document]
Comment (tomsmyers) [reply server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:103]: good questions both. on the first yes - here is the ticket https://dsdmoj.atlassian.net/browse/IC-1412

on the second, i'd rather not burn more time on it right now if possible.

File: server/utils/viewUtils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply server/utils/viewUtils.ts:67]: Cross-posting from Slack, for future reference: 

> Yeah I’d be keen to keep it as is - don’t really fancy writing our own versions of those macros. I think it’s easy with the tag one to say that the HTML is simple, but for some of the more complex components (and things get complex quickly once you start needing to be able to display e.g. input fields in error states) we really don’t want to be replicating that stuff that’s in the GOV.UK Frontend library.
>
> I think the options are either:
> - pass the macros from template back to view class (what we do now, a little clunky but works)
> - directly pull the macros out of the Nunjucks environment (we do something similar here https://github.com/ministryofjustice/hmpps-interventions-ui/blob/main/server/utils/viewUtils.ts#L7 but would need to see how that works with macros that you need to import)
>
> So I’d say it’s best to stick with doing the first one for now and maybe create a tech debt to look at the latter.

[the remark this answers is not in the document]
Comment (tomsmyers) [reply server/utils/viewUtils.ts]: this is now changed

**Review by Gweaton — APPROVED** [review]
Have tidied up the commits on this as discussed on Slack. Looks good to me!

**Pull request merged**